feat(IT-14): Implement user registration page

qiningliu1 opened the pull request:
###  Features Implemented:
- Created a **static frontend page** for the **user sign-up and sign-in process** via the email pathway.
- Implemented form elements and validation following the requirements.


###  UI & Layout:
- Implemented the following form fields:
  - **User Name** (text input)
  - **Email Address** (email input, required)
  - **Password** (password input, required)
  - **Confirmed Password** (password input, required)
  - **Language** (dropdown/select input)
  - **phone number** (text input)
  - **Self Description** (text area)

###  Styling & Responsiveness:
- Used **Chakra UI** for consistent styling.
- Ensured the page is **fully responsive and mobile-friendly**.
- Added **error messages** for required fields and validation failures (e.g., invalid email format, weak password).

###  Form Validation & User Experience:
- Implemented **basic frontend validation** to ensure required fields are filled out.
- Used **`react-hook-form` with `zod`** for schema validation.
- Displayed **error messages** for validation failures.

###  Technical Notes:
- Ensured form styles are **consistent with the overall design** of the application.
- **Backend integration (e.g., sending form data)** will be handled in a separate task.


### Notes for Dependencies*
- Considering version for chakra-ui/react above 3 is not fully compatible withe form-control and select component
- the @chakra-ui/form-control and @chakra-ui/select should be imported separately


Related task: IT-14


### Static page comparative with UI


![static_deifference](https://github.com/user-attachments/assets/bbb11fe2-7bfe-4248-ac8c-0720cffe3e23)



### Input comparative with UI

![input_compare](https://github.com/user-attachments/assets/047b422c-7bdd-40d5-b811-66e3a97a28f7)


### DashBoard responsive signup


![present2](https://github.com/user-attachments/assets/fa678212-9feb-4ee6-8860-382a56b39a4c)


Responsive-design-mobile

![present3](https://github.com/user-attachments/assets/cbd5b13d-cc5a-4d4b-b7aa-704239bb7774)


Hongyuan Wang - bug fix:

1.  All useless comments removed in code
2. src/module/components.tsx was moved to src/common folder to make sure it can be shared to every components
3. A part of this file src/app/page.tsx named "dashboard" has changed to a better name
4. src/app/page.tsx is for our landing page, it should not be a place for putting sign up page, I move this page main function to src/app/signup/page.tsx
5. move Zod schema to another folder and make sure other component can share this Zod schema.

Chris-Wang reviewed (changes requested):
Did not update as what we discussed yesterday:
- page.tsx needs to reduce to less than 50 lines
- sub folder needed in module

qiningliu1 commented:
Update new file routing:

-Split components in sign-up-form 
-Reduce page.tsx less then 50 lines

StevenRen929 committed:
feat/IT-19-Third-party-sign-up-and-login
-add third party login / signup function for google button on sign up page
-solve the refresh lose token error by using useEffect and getItem from the localstorage

resolve it-19

feat/IT-19-Third-party-sign-up-and-login
update
-add user sign out to remove user token from local stroagre
-add sign out Loading animation

update it-19

feat/IT-19-Third-party-sign-up-and-login

update:
-add user fail third party auth page and add redirction to signup page
-remove useless comments

resolve IT-18

feat/IT-19-Third-party-sign-up-and-login
update
-remove useless if(window!='undefined') in useEffect , these are code used to debug hydration error at the beginning

update IT-19

feat/IT-19-Third-party-sign-up-and-login
Update
-Do modularization for dashbord and oauth error page
-handle Oauth Error by change the state of signup page via url params to improve security
-delete the useless comments

Update IT-19

feat/IT-19-Third-party-sign-up-and-login
-update remove all url params if oauth fail

update IT-19

Update
-implement a route for auth callback in front end, using getUserProfile API call in front end to do validation for token

-Apply Error Context for Auth Error Handling

Update IT-19

Update
-move ErrorContext Global provider to Context folder from lib folder

feat/IT-19-Third-party-sign-up-and-login
Update
-Remove console.log in login-callback
-apply getUserProfile api call in dashboard to minimize code redundance

Bugfix IT-19

update
-remove console log

update it-19

Bugfix
--Apply suspense to resolve npm run build error

bugfix it-19

UPDATE
-add prettier to the project
-add npm run format  in package.json to run  prettier formatter any time
-modularize ThirdPartyFailureAction in previous auth-failure-handler in module folder in order for easier update in new updated UI(failure of email also need to be considered in the future and will be used in auth-error-handler by using states in the future)
-update errorMessage in lib replace CREDENTIAL_ERROR by FAIL_THIRD_PARTY_AUTHENTICATION_ERROR

update IT-19

Update
-implement husky
-test husky pre commit function
-implement lint-state in package.json
-implement prettier in lint-state

Note: Testing 1

update IT-19

update

Note: Test pre-commit-2

Update
-test pre commit 3

update
-test echo message pre-submit
-test pre-commit

Note: Test 4

update IT-19

update
-change script:{
prepare:{hasky} to prepare:{hasky install}
}
-change pre-commit message
-Test  precommit

update IT-19

qiningliu1 committed:
feat(IT-14): User Sign-Up - Designed and updated signup page UI   - Improved responsive design for mobile   - Added Username and Email and phone number verification   - Applied zodResolved and Chakra-UI for form validation   - language selection feature stil need to update   - Add dashboard responsive mode

Resolve IT-14 - Qining Liu

fix: restore provider.tsx and other local updates

update: pull update from dev and fix conflict

Hongyuan Wang - bug fix:

1.  All useless comments removed in code
2. src/module/components.tsx was moved to src/common folder to make sure it can be shared to every components
3. A part of this file src/app/page.tsx named "dashboard" has changed to a better name
4. src/app/page.tsx is for our landing page, it should not be a place for putting sign up page, I move this page main function to src/app/signup/page.tsx
5. move Zod schema to another folder and make sure other component can share this Zod schema.

Co-Authored-By: qiningliu1 <101502608+qiningliu1@users.noreply.github.com>

qiningliu1 commented:
Update revise
-Background colour setting
-Password icon
-bottom button space uniform

Chris-Wang reviewed (changes requested):
<img width="1701" alt="image" src="https://github.com/user-attachments/assets/2e261fc0-838b-4601-91d7-8b020d8a546d" />
<img width="1707" alt="image" src="https://github.com/user-attachments/assets/1af7d552-5665-489e-9a4b-af614aacdd61" />
run it in chrome and safari:
- still have black background 
- input label could is hardly readable
- consider to add desktop mode, current these the mobile and desktop looks the same

Chris-Wang commented:
I can still see issues on the left bottom corner

qiningliu1 commented:
Update revise
-update global background color setting at provider and layout file
-update label color from gran300 to 600
-add responsive dashboard design
-reviese chrome conflict and nextJS warning message

Chris-Wang commented on src/app/page.tsx:
```diff
@@ -6,17 +6,56 @@
+        {/* Right dashboard：dashboard show，mobile not show */}
+        <Box
+          flex="1"
+          // display={{ base: "none", md: "block" }}
```
commented code needs to be removed

Chris-Wang commented on src/module/components.tsx:
```diff
@@ -0,0 +1,29 @@
+// src/module components.tsx
+
+export const inputStyles = {
```
please format this file to be more readable

Chris-Wang commented on src/module/sign-up-form/confirm-password.tsx:
```diff
@@ -0,0 +1,80 @@
+          {showConfirmPassword ? (
+            <FiEye />
+          ) : (
+            // <Image
```
commented code needs to be removed

Chris-Wang commented on src/module/sign-up-form/email.tsx:
```diff
@@ -0,0 +1,54 @@
+  FormErrorMessage,
+} from '@chakra-ui/form-control'
+import { UseFormRegister, FieldErrorsImpl } from 'react-hook-form';
+// Import the styles for the form inputs and buttons
```
no comment need for this one

Chris-Wang commented on src/module/sign-up-form/password.tsx:
```diff
@@ -0,0 +1,75 @@
+
+import { SignupFormData } from './sign-up-form';
+
+interface PasswordFieldProps {
```
format this file please

Chris-Wang commented on src/module/sign-up-form/phone-number.tsx:
this file needs to be formatted too

Chris-Wang commented on src/module/sign-up-form/sign-up-form.tsx:
this file needs to be formatted too

Chris-Wang commented on src/module/sign-up-form/sign-up-form.tsx:
```diff
@@ -0,0 +1,218 @@
+import { PhoneNumberField } from './phone-number';
+import { DescriptionField } from './description';
+// 1. Definite the schema for the form using Zod
+const signupSchema = z
```
validation schema should be moved to a sharable place

Chris-Wang commented on src/app/page.tsx:
```diff
@@ -6,17 +6,56 @@
-  
-  // This return value will never be rendered because of the redirect
-  return null
+  return (
```
as we mentioned earlier, those styled components should be moved to module, to leave page.tsx more clear

Chris-Wang commented on src/app/page.tsx:
```diff
@@ -6,17 +6,56 @@
+    <Box minH="100vh" bg="white">
+      <Flex
+        direction={{ base: "column", md: "row" }}
+        //dashboard stable width and height, mobile adaptable
```
I don't think it should be called as `dashboard`, we have a page called `dashboard` which is used as home page after user login

Chris-Wang commented on src/module/components.tsx:
if this file is only used for storing style of button or input, it should be put in common/button.tsx and common/formInput.tsx

Chris-Wang commented on src/app/signup/page.tsx:
refactor this page, to something like
```
export default function Page() {
  return <SignUp />;
}
```

Chris-Wang commented on src/app/page.tsx:
page.tsx is revered for our landing page, it should not be a place for putting sign up page here

qwewhy committed:
feat(IT-14): User Sign-Up - Designed and updated signup page UI

IT19 rebase in this IT14

Co-Authored-By: qiningliu1 <101502608+qiningliu1@users.noreply.github.com>

qiningliu1 committed:
update fix:
-Top navigation space
-Password error message color
-Language still not fixed because some charka error
-left form height restriction
